server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -194,14 +194,27 @@ public IndexNameExpressionResolver(Settings settings) {
             }
 
             Collection<IndexMetaData> resolvedIndices = aliasOrIndex.getIndices();
-            if (resolvedIndices.size() > 1 && !options.allowAliasesToMultipleIndices()) {
-                String[] indexNames = new String[resolvedIndices.size()];
-                int i = 0;
-                for (IndexMetaData indexMetaData : resolvedIndices) {
-                    indexNames[i++] = indexMetaData.getIndex().getName();
+            if (resolvedIndices.size() > 1 && options.allowAliasesToMultipleIndices() && options.requireAliasesToWriteIndex()) {
```

Comment:
Indeed, I think we should maintain behavior and throw an exception if resolvedIndices.size() > 1 and options. requireAliasesToWriteIndex() == false and options.allowAliasesToMultipleIndices()  == false too. Also can you assert in the indexOptions class that options.allowAliasesToMultipleIndices() and requireAliasesToWriteIndex() are never used together? it's confusing.